From ef15d33b3c99287a07d3e9725bf45aafb0f27000 Mon Sep 17 00:00:00 2001
From: Suresh Gupta <suresh.gupta@xilinx.com>
Date: Fri, 5 Oct 2018 19:42:54 +0530
Subject: [PATCH 0447/1852] v4l: xilinx-multi-scaler: Return the proper
 parameters in S_FMT

commit 7fcc49536461e7030206031bbd9ae09da2b5abc2 from
https://github.com/Xilinx/linux-xlnx.git

The VIDIOC_S_FMT ioctl returns the current format parameters as
VIDIOC_G_FMT does.

Signed-off-by: Suresh Gupta <suresh.gupta@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Reviewed-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-multi-scaler.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-multi-scaler.c b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
index 150937c56401..e95a1acd4fa8 100644
--- a/drivers/media/platform/xilinx/xilinx-multi-scaler.c
+++ b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
@@ -1121,7 +1121,9 @@ vidioc_s_fmt(struct xm2msc_chan_ctx *chan_ctx, struct v4l2_format *f)
 
 	for (i = 0; i < q_data->nplanes; i++) {
 		q_data->bytesperline[i] = q_data->stride;
+		pix->plane_fmt[i].bytesperline = q_data->bytesperline[i];
 		q_data->sizeimage[i] = q_data->stride * q_data->height;
+		pix->plane_fmt[i].sizeimage = q_data->sizeimage[i];
 	}
 
 	xm2msc_pr_q(chan_ctx->xm2msc_dev->dev, q_data,
-- 
2.31.1

